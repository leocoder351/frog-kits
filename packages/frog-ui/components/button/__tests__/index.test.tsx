import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from '../index';

describe('Component <Button /> Test', () => {
  let testButtonClicked = false;
  const onClick = () => {
    testButtonClicked = true;
  };

  test('should render default', () => {
    // snapshot test
    const component = renderer.create(<Button onClick={onClick}>default</Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // dom test
    render(<Button onClick={onClick}>default</Button>);
    const btn = screen.getByText('default');
    fireEvent.click(btn);
    expect(testButtonClicked).toEqual(true);
  });

  test('should render specific type', () => {
    const types: any[] = ['default', 'primary', 'danger', 'warning'];
    const component = renderer.create(
      <>
        {types.map((type) => (
          <Button key={type} type={type} onClick={onClick}>
            {type}
          </Button>
        ))}
      </>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
