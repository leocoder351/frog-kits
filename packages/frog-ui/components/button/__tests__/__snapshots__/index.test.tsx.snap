// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <Button /> Test should render default 1`] = `
<button
  className="frog-btn frog-btn-default"
  onClick={[Function]}
>
  default
</button>
`;

exports[`Component <Button /> Test should render specific type 1`] = `
Array [
  <button
    className="frog-btn frog-btn-default"
    onClick={[Function]}
  >
    default
  </button>,
  <button
    className="frog-btn frog-btn-primary"
    onClick={[Function]}
  >
    primary
  </button>,
  <button
    className="frog-btn frog-btn-danger"
    onClick={[Function]}
  >
    danger
  </button>,
  <button
    className="frog-btn frog-btn-warning"
    onClick={[Function]}
  >
    warning
  </button>,
]
`;
